src/registrar/models/domain.py

```diff
@@ -1171,34 +1188,47 @@ def _delete_domain(self):  # noqa
 
         logger.info(f"Finished deleting contacts for {self.name}")
 
-        # delete ds data if it exists
+    def _delete_dnssecdata(self):
+        """Removes DNSSEC data"""
         if self.dnssecdata:
             logger.debug("Deleting ds data for %s", self.name)
             try:
                 # set and unset client hold to be able to change ds data
-                logger.info("removing client hold")
+                logger.info("Removing client hold")
                 self._remove_client_hold()
                 self.dnssecdata = None
-                logger.info("placing client hold")
+                logger.info("Placing client hold")
                 self._place_client_hold()
             except RegistryError as e:
                 logger.error("Error deleting ds data for %s: %s", self.name, e)
                 e.note = "Error deleting ds data for %s" % self.name
                 raise e
 
-        # check if the domain can be deleted
-        if not self._domain_can_be_deleted():
-            note = "Domain has associated objects that prevent deletion."
-            raise RegistryError(code=ErrorCode.COMMAND_FAILED, note=note)
+    def _delete_related_objects_from_db(self):
```

Comment:
Nice work on cleaning up the deletion logic! Tested and it works. One potential refactoring option to consider: we could leverage Django's CASCADE relationships to make this even cleaner. And we could tie everything possibly back to the house so then using cascade, we could just do a self.hosts.all().delete() . This would require changing the foreign key relationships... Flagging it as a potential for consideration. The current implementation works great for the immediate requirements! 👍 it really depends on the level of scope for refactoring.

Reply:
After discussing with Rebecca, it seems the scope for the ticket was met. I do feel that we should be handling relationships and clean deletions by using cascade, which would encompass changes to our database layer. I will bring it up separately but approving lesson since it's meeting the scope previously discussed (which I was nto privi too) and it works.